src/hotspot/share/ci/ciArrayKlass.cpp

```diff
@@ -103,3 +104,17 @@ ciArrayKlass* ciArrayKlass::make(ciType* element_type) {
     return ciObjArrayKlass::make(element_type->as_klass());
   }
 }
+
+GrowableArray<ciInstanceKlass*>* ciArrayKlass::interfaces() {
```

Comment:
FTR there's a subtle asymmetry between `ciArrayKlass::interfaces()` and `ciInstanceKlass::transitive_interfaces()` when it comes to memory allocation: the former allocates from resource area while the latter from compiler arena.

It doesn't cause any problems since `ciArrayKlass::interfaces()` is used only in `Type::Initialize_shared()` to instantiate shared `TypeAryPtr::_array_interfaces`, but it took me some time to find that out. 

Maybe a helper method in `type.cpp` is a better place for that logic.

Reply:
How would that work? ciArrayKlass::interfaces() has to transition into the vm which is not something that would feel right in type.cpp.

Reply:
Indeed, good point.

IMO you could just work directly with CI mirrors for `Serializable` and `Cloneable`, but now I'm curious why does CI code diverge from `ObjArrayKlass::compute_secondary_supers()`?
  https://github.com/openjdk/jdk/blob/master/src/hotspot/share/oops/objArrayKlass.cpp#L376

Reply:
And to answer my question: only `Serializable` and `Cloneable` are interfaces among those. The rest are arrays.